File: Client/src/app/models/dtos/album.dto.ts
import { AlbumType } from "../enums/album-type.enum";
import { ArtistDto } from "./artist.dto";
import { SongDto } from "./song.dto";

export interface AlbumDto {
    id: string,
    name: string,
    cover: string,
    year: number,
    albumType: AlbumType,
    artist: ArtistDto,
    songs: SongDto[],
}

File: Tabloid.Domain/DataTransferObjects/AlbumDto.cs
using Tabloid.Domain.Enums;

namespace Tabloid.Domain.DataTransferObjects
{
    public class AlbumDto : IDto<Guid>
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public string Cover { get; set; }

        public int Year { get; set; }

        public AlbumType AlbumType { get; set; }


        public ArtistDto Artist { get; set; }

        public ICollection<SongDto> Songs { get; set; }
    }
}
